client/src/contexts/NotificationContext.js

```diff
@@ -0,0 +1,38 @@
+import React, { createContext, useState } from 'react'
+
+export const NotificationContext = createContext({})
+
+export const NotificationContextProvider = ({ children }) => {
+  const [notifications, setNotifications] = useState([
+    {
+      test: {
+        id: 'test',
+        type: 'success',
+        message: 'this is fun!'
+      }
+    }
+  ])
+
+  const showNotification = (id, message, type) =>
+    setNotifications((prev) => [
+      ...prev,
+      {
+        [id]: {
+          id,
+          message,
+          type
+        }
+      }
+    ])
```

Comment:
```suggestion
  const showNotification = (message, id, type = 'success') =>
    setNotifications((prev) => [
      ...prev,
     {
     id: id || Date.now()
      message,
      type
    }
    ])
```

I would recommend switching the order